### PR TITLE
MMT fixes

### PR DESCRIPTION
**Pull request by mmatczuk**
Code cleanup. It would be good to see a diff against master as it removes some of the custom code. I'm pretty sure that filling needs fixing.

### CODE REVIEW

**Comment by dahankzter** [comment]
`func (pool *hostConnPool) Close()` should probably close the sharded connections too.

File: connectionpool.go [hunk 1]
```diff
@@ -278,72 +276,78 @@ func (h *hostConnPool) String() string {
...
 
+	si := pool.host.ShardingInfo()
+	sharded := si.nrShards > 0 && token != nil
+
+	poolSize := pool.size
+	if sharded {
+		poolSize = int(si.nrShards)
+	}
+
 	size := len(pool.conns)
```
Comment (dahankzter) [inline connectionpool.go]: This can be wrong if we are sharded?

**Comment by dahankzter** [comment]
`func (pool *hostConnPool) fill()` needs to handle the sharded case as well.

File: connectionpool.go (hunk 1 above)
Comment (mmatczuk) [reply connectionpool.go]: No they are in sync see in connect.

Comment (dahankzter) [reply connectionpool.go]: Ah it's fine now that I got that the map is just an index. It needs to be kept in sync though.

**Pull request merged**